src/main/java/Control/Scenario/UC_CreateFlat.java

```diff
@@ -0,0 +1,211 @@
+                dict.put(T_FlatOwner.DBNAME_MIDDLENAME, firstTimeCreation.getOwner2().getMiddleName());
+                dict.put(T_FlatOwner.DBNAME_LASTNAME, firstTimeCreation.getOwner2().getLastName());
+                dict.put(T_FlatOwner.DBNAME_PHONE, firstTimeCreation.getOwner2().getPhone());
+                dict.put(T_FlatOwner.DBNAME_EMAIL, firstTimeCreation.getOwner2().getEmail());
+                dict.put(T_FlatOwner.DBNAME_ADDRESS, firstTimeCreation.getOwner2().getAddress());
+
+                I_FlatOwner.insert(db.getConn(), db.getPs(), T_FlatOwner.CreateFromScratch(dict));
+
+                newlyInsertedFlatOwner2 = I_FlatOwner.retrieveLatestPerConnectionInsertedID(db.getConn(), db.getPs(), db.getRs());
+
+                dict = new Hashtable();
+
+                dict.put(T_FlatOwner_flat.DBNAME_FLATOWNERID, newlyInsertedFlatOwner2);
+                dict.put(T_FlatOwner_flat.DBNAME_FLATID, newlyInsertedFlatId);
+
+                I_FlatOwner_flat.insert(db.getConn(), db.getPs(), T_FlatOwner_flat.CreateFromScratch(dict));
+
+            }
+
+            /*  */
+            // Create new ControllerUnit
+            dict = new Hashtable();
+
+            dict.put(T_ControllerUnit.DBNAME_UID, firstTimeCreation.getUid());
+            dict.put(T_ControllerUnit.DBNAME_DIPADDRESS, firstTimeCreation.getDip());
+            dict.put(T_ControllerUnit.DBNAME_ZWAVE, firstTimeCreation.getZwave());
+            dict.put(T_ControllerUnit.DBNAME_CENTRALUNIT_ID, firstTimeCreation.getCentralUnitId());
+            dict.put(T_ControllerUnit.DBNAME_FLAT_ID, newlyInsertedFlatId);
+
+            I_ControllerUnit.insert(db.getConn(), db.getPs(), T_ControllerUnit.CreateFromScratch(dict));
+
+            // Success
+            db.afterOkSqlExecution();
+
+            jsonResponse.setStatus(HttpServletResponse.SC_CREATED);
+            jsonResponse.setMessage("Flat + FlatOwner(s) and Controller created.");
+        } catch (SQLException e) {
+            db.afterExceptionInSqlExecution(e);
+            jsonResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            jsonResponse.setMessage("Internal server error.");
+
+        } catch (CreationException e) {
+            db.afterExceptionInSqlExecution(e);
+            jsonResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            jsonResponse.setMessage("Testing");
+        }
+
+        return jsonResponse;
+    }
+
+    private boolean dipControllerAlreadyPresentForCentralUnit(int centralUnitId, String dip) throws SQLException {
+        List<T_ControllerUnit> list = I_ControllerUnit.retrieveFilteredAll(db.getConn(), db.getPs(), db.getRs(), centralUnitId, DB_DO_NOT_USE_THIS_FILTER, dip);
+
+        if (list.size() > 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+
+    // privates
+    private boolean everythingOkForCreation(@NotNull final FlatFirstTimeCreation firstTimeCreation) throws NumberFormatException {
```

Comment:
everything? co je to?

Reply:
ved vsetky atributy z creationu flatu :D mohol tam dat dataOkForCreation a bolo by to to iste